models/activities/action.go

```diff
@@ -393,10 +393,13 @@ func (a *Action) GetCreate() time.Time {
 	return a.CreatedUnix.AsTime()
 }
 
-// GetIssueInfos returns a list of issues associated with
-// the action.
+// GetIssueInfos returns a list of associated information with the action.
 func (a *Action) GetIssueInfos() []string {
-	return strings.SplitN(a.Content, "|", 3)
+	ret := strings.SplitN(a.Content, "|", 3)
+	for len(ret) < 3 {
+		ret = append(ret, "")
+	}
```

Comment:
Do we know in 2 years why that is needed? The whole delimiter concept is fragile because the values are not just numbers but user provided text which can contain the delimiter:
https://github.com/go-gitea/gitea/blob/a2b0fb1a64b0794b808a013089758a49f56d8915/services/feed/action.go#L232
https://github.com/go-gitea/gitea/blob/a2b0fb1a64b0794b808a013089758a49f56d8915/services/feed/action.go#L278
https://github.com/go-gitea/gitea/blob/a2b0fb1a64b0794b808a013089758a49f56d8915/services/feed/action.go#L310
That field should be json or a struct which can be json serialized by xorm. Here for example it contains json text:
https://github.com/go-gitea/gitea/blob/a2b0fb1a64b0794b808a013089758a49f56d8915/services/feed/action.go#L344

Reply:
I'd like don't show the comment which not have enough data.

Reply:
> I'd like don't show the comment which not have enough data.

Then it needs a lot of new `{{if}}` blocks in the templates. I don't think it's worth to make it that complex.